docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,120 @@
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Derived Types](./learning.md#derived-types), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) Base types representing the shape of domain models should be in the domain directory at the bounded context root. These types can be shared, often for the purpose of derivatives, among slices.
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) Domain types, regardless of whether scoped to a slice or scoped to a bounded context, should not depend on code that is not itself either domain code or is a generic utility type (error handling, ts utils, and such). *This is a concept borrowed from Clean/Hexagonal/Onion architecture and key piece in the internal consistency of our domain model if followed. Exceptions to this rule should be very slim or non existent.*
+
+- [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Repository pattern](./learning.md#repository-pattern), [UA: Persistence Agnosticism](./learning.md#persistence-agnosticism) Repository layer code is the only code that should depend on types representing the shape of database table/view/query/etc. *The implementation details of data persistence should not be exposed to consumers of a domain model that utilizes said persistence. One of the quickest ways to a building a ball of mud is to do let a database schema (slow and risky to change) be the source of truth for the shape of a domain model across all layers and systems. There are cases where this is ok (constrained problem space, prototypes, etc), but this should be avoided by default.*
+
+- [UA: Slim APIs are Easier to Maintain](./learning.md#slim-apis-are-easier-to-maintain) Don't require more data for a function than is required. *A slim API is less risky to change and easier maintain and reason about.*
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Eventual Consistency](./learning.md#eventual-consistency) All data should only have a single source of truth. *It is ok for data to be stored in multiple places as long as the source of truth is always one location. For concepts that exist within multiple domains (students), it is perfectly acceptable for that concept to have it's source of truth spread across data stores as long as no part (field) of any individual meaning is has it's source of truth in more than one location.*
```

Comment:
```suggestion
- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Eventual Consistency](./learning.md#eventual-consistency) All data should only have a single source of truth. *It is ok for data to be stored in multiple places as long as the source of truth is always one location. For concepts that exist within multiple domains (students), it is perfectly acceptable for that concept to have its source of truth spread across data stores as long as no part (field) of any individual meaning is has its source of truth in more than one location.*
```

Reply:
Would it be clearer to change the last sentence to:

"... of any individual meaning is stored in more than one location."